bot/exts/info/tags.py

```diff
@@ -373,38 +303,68 @@ async def get_command(self, ctx: Context, *, argument_string: Optional[str]) ->
         Returns True if a message was sent, or if the tag is on cooldown.
         Returns False if no message was sent.
```

Comment:
>I think we should try to avoid any cases of "The application did not respond" errors on the client. Another case where this can happen is where no match can be found e.g. /tag get fionferonfero

@wookie184 Should it always return `True` now as an embed has to be sent either ways or should it return `False` when no match for a tag is found?

Reply:
We need to be careful to ensure that we don't break normal tag invocation (with just the name of the tag, e.g. `!code`). This is handled in the error handler here https://github.com/python-discord/bot/blob/main/bot/exts/backend/error_handler.py#L162

For that it makes sense to give no result if a tag wasn't found, although if a user is using the `tag get` command explicitly I think they should always get a response (e.g. saying "no tag found").

I'm not sure what the best way to make this change is, it might require changes to the error handler (maybe making it use a helper function which is also used by the get command, but the get command can also handle no tags found).

Reply:
I think it gets my question unanswered, my question was that earlier the `tag` text command function returned `False` when no embed was sent but as of now the `tag` slash command sends an embed either way(if the tag was found and if the tag wasn't found). So, should the function now return `True` if an embed is sent for no tag found or should it return `False` when a tag is not found. 

Also about the error handler, I have already made changes to it i.e migrated it to handle slash command, thats why the tests even work.

Reply:
Also wookie the `!code` is already broken as the tag command now uses `discord.Interaction` and the ctx captured with the message containing `!code` has its interaction attribute return `None`, which means the `tag` command can't be invoked. A possible work around for this would be to have `get_command_ctx` in `tags.py` and `try_get_tag_ctx` in `error_handler.py` or we have to make the `get_command` ugly by doing `isinstance` everywhere.

Reply:
(sorry i'm sure you know this already, I just wanted to summarise to ensure we're on the same page).

The error handler is used as the individual tag names aren't registered as commands (e.g. `!code`), so we catch any CommandNotFound errors, and in those cases try and match the name given to a tag (this also allows us to do fuzzy matching that doesn't interfere with other tags).

Currently the `!tag get <name>` command, and direct `!<name>` invocation are pretty much exactly the same and use the same code. This has a few issues:
- It means the behaviour has to be the same (e.g. with behaviour if no tag is found, with `!tag get` we'd want to say that but with `!<name>` we'd want to be silent)
- It is now an issue since `/tag get <name>` is an interaction while `!<name>` isn't.

Because of this, as you say, it would probably be easiest to make the code for the direct invocation separate. There are a couple of different ways we could do this. One would be to have a `def search_tags_by_name(self, name: str)` function in the `Tags` cog, this can potentially be used by the interaction code and also in the error handler (by getting the cog using `bot.get_cog`), which could then handle what to do with the tags found itself. (that's not a requirement that you implement it like that at all, you can see what works best as you're doing it)

Reply:
Finally we are on the same page👍🏽